PSReadLine/ReadLine.cs

```diff
@@ -625,8 +625,7 @@ private PSConsoleReadLine()
             {
                 try
                 {
-                    ps.AddCommand("Get-Variable").AddParameter("Name", "host").AddParameter("ValueOnly");
-                    var results = ps.Invoke();
+                    var results = ps.AddScript("$Host").Invoke();
```

Comment:
I think if we used `Invoke<PSHost>`, we could avoid the dynamic code below.

Reply:
Agreed! Will change it right away.